llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -265,6 +275,72 @@ void collectCompositeElementsInfoRecursive(
   }
 }
 
+/// Recursively iterates over a composite type in order to collect information
+/// about default values of its scalar elements.
+/// TODO: processing of composite spec constants here is similar to
+/// collectCompositeElementsInfoRecursive. Possible place for improvement -
```

Comment:
Is it also similar to emitSpecConstantRecursiveImpl and generateSpecConstDefaultValueMetadata?

Reply:
Not really, it doesn't look similar. `generateSpecConstDefaultValueMetadata` is not related to recursive methods, and `emitSpecConstantRecursiveImpl` is too different from this function.